polygeist/tools/cgeist/Lib/clang-mlir.cc

```diff
@@ -4135,50 +4135,30 @@ static bool parseMLIR(const char *Argv0, std::vector<std::string> filenames,
   const char *binary = Argv0; // CudaLower ? "clang++" : "clang";
   const std::unique_ptr<Driver> driver(
       new Driver(binary, llvm::sys::getDefaultTargetTriple(), Diags));
-  std::vector<const char *> Argv;
+  OptionsPasser Argv;
   Argv.push_back(binary);
-  for (auto a : filenames) {
-    char *chars = (char *)malloc(a.length() + 1);
-    memcpy(chars, a.data(), a.length());
-    chars[a.length()] = 0;
-    Argv.push_back(chars);
+  for (const auto &filename : filenames) {
+    Argv.push_back(filename);
   }
   if (FOpenMP)
     Argv.push_back("-fopenmp");
   if (TargetTripleOpt != "") {
-    char *chars = (char *)malloc(TargetTripleOpt.length() + 1);
-    memcpy(chars, TargetTripleOpt.data(), TargetTripleOpt.length());
-    chars[TargetTripleOpt.length()] = 0;
     Argv.push_back("-target");
-    Argv.push_back(chars);
+    Argv.push_back(TargetTripleOpt);
   }
   if (McpuOpt != "") {
-    auto a = "-mcpu=" + McpuOpt;
-    char *chars = (char *)malloc(a.length() + 1);
-    memcpy(chars, a.data(), a.length());
-    chars[a.length()] = 0;
-    Argv.push_back(chars);
+    Argv.emplace_back("-mcpu=", McpuOpt);
   }
   if (Standard != "") {
-    auto a = "-std=" + Standard;
-    char *chars = (char *)malloc(a.length() + 1);
-    memcpy(chars, a.data(), a.length());
-    chars[a.length()] = 0;
-    Argv.push_back(chars);
+    Argv.emplace_back("-std=", Standard);
   }
   if (ResourceDir != "") {
     Argv.push_back("-resource-dir");
-    char *chars = (char *)malloc(ResourceDir.length() + 1);
-    memcpy(chars, ResourceDir.data(), ResourceDir.length());
-    chars[ResourceDir.length()] = 0;
-    Argv.push_back(chars);
+    Argv.push_back(ResourceDir);
```

Comment:
Argv.emplace_back("-resource-dir ", ResourceDir);

Reply:
This would imply unneeded copying.